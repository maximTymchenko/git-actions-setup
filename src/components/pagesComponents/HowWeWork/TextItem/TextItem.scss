@import '../../../../styles/vars';

.text-item {
    &__wrapper {
        display: flex;
        justify-content: space-between;
        margin-top: 100px;
        align-items: center;

        @media screen and (max-width: $desktop) {
            flex-wrap: wrap;
            justify-content: center;
            margin-bottom: 50px;
        }
        &:last-child {
            flex-direction: row-reverse;
            justify-content: space-between;

            @media screen and (max-width: $desktop) {
                flex-wrap: wrap;
                justify-content: center;
            }

            @media screen and (max-width: $mobile) {
                flex-direction: column-reverse;
            }
        }

        @media screen and (max-width: $mobile) {
            margin-top: 70px;
            flex-direction: column-reverse;
        }
    }

    &__desc {
        font-family: 'OktaNeue-Regular';
        font-size: 15px;
        color: $text__body;
        margin-bottom: 30px;
    }

    &__img {
        margin-right: 35px;

        //! test small container
        &--small {
            max-width: 500px;
            max-height: 450px;
        }

        @media screen and (max-width: $desktop) {
            margin-right: 0;
        }

        @media screen and (max-width: $tablet) {
            margin-top: 40px;
            display: block;
            width: 100%;
            max-width: 740px;
        }

        @media screen and (max-width: $mobile) {
            margin: 0px 0 60px;
            display: block;
            width: 100%;
            max-width: 350px;
        }
    }
}
