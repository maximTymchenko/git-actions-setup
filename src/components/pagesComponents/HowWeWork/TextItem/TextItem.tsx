import React, { FC } from 'react';
import SectionTitle from '../../../common/SectionTitle';
import Button from '../../../common/Button';
import arrowRight from '../../../../assets/img/arr-right.svg';
import './TextItem.scss';
import { Link } from 'gatsby';

interface ITextItem {
    itemTitle: string;
    itemDesc: string;
    image: any;
    link: string;
    page: string;
    size: string;
}

const TextItem: FC = (props: ITextItem) => {
    return (
        <div data-testid="text-item" className="text-item__wrapper">
            <div className="text-item__body">
                <SectionTitle type={'primary'} title={props.itemTitle} />
                <p className="text-item__desc">{props.itemDesc}</p>
                <Link
                    className="link"
                    to={props.link}
                    activeClassName="header__nav-text header__nav-text--active"
                    state={{ choice: props.page }}
                >
                    <Button
                        icon={arrowRight}
                        type={'primary'}
                        className="button"
                        desc={'Learn more'}
                    >
                        Learn more
                    </Button>
                </Link>
            </div>
            <img
                src={props.image}
                alt=""
                className={`text-item__img text-item__img--${props.size}`}
            />
        </div>
    );
};

export default TextItem;
