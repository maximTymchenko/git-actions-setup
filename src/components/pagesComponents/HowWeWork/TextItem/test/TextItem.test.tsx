import React from 'react';
import TextItem from '../TextItem';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(<TextItem itemDesc="test" itemTitle="test"></TextItem>);
    expect(getByTestId('text-item')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<TextItem itemDesc="test" itemTitle="test"></TextItem>).toJSON();
    expect(tree).toMatchSnapshot();
});
