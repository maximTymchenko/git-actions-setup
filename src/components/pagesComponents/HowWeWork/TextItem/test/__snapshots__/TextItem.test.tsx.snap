// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="text-item__wrapper"
  data-testid="text-item"
>
  <div
    className="text-item__body"
  >
    <h2
      className="title title--primary"
      data-testid="section-title"
    >
      test
    </h2>
    <p
      className="text-item__desc"
    >
      test
    </p>
    <a
      className="link"
    >
      <button
        className="button button--primary button--undefined button--undefined"
        data-testid="button"
      >
        Learn more
        <img
          alt="Learn more"
          className="button__icon button__icon--undefined"
          data-testid="button__icon"
          src="test-file-stub"
        />
      </button>
    </a>
  </div>
  <img
    alt=""
    className="text-item__img text-item__img--undefined"
  />
</div>
`;
